### PR TITLE
Storage validation of hosts.

### PR DESCRIPTION
**Pull request by JumaX**
# Description

This PR adds and additional check to validate that there is enough free disk space across all mounts on all hosts listed in the inventory file, to install and run the product.

Fixes # (issue)

ANSIENG-525.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

This has been validated by running the plaintext-rhel scenarion and confirming that all mount points on each host are validated has having at least 70% free space.

eg:

`ok: [kafka-rest1] => (item={'block_used': 8331220, 'uuid': 'N/A', 'size_total': 270294650880, 'block_total': 65989905, 'mount': '/etc/resolv.conf', 'block_available': 57658685, 'size_available': 236169973760, 'fstype': 'ext4', 'inode_total': 16777216, 'options': 'rw,relatime,bind', 'device': '/dev/vda1', 'inode_used': 275592, 'block_size': 4096, 'inode_available': 16501624}) => {
    "ansible_loop_var": "item",
    "changed": false,
    "item": {
        "block_available": 57658685,
        "block_size": 4096,
        "block_total": 65989905,
        "block_used": 8331220,
        "device": "/dev/vda1",
        "fstype": "ext4",
        "inode_available": 16501624,
        "inode_total": 16777216,
        "inode_used": 275592,
        "mount": "/etc/resolv.conf",
        "options": "rw,relatime,bind",
        "size_available": 236169973760,
        "size_total": 270294650880,
        "uuid": "N/A"
    },
    "msg": "All assertions passed"
}`

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] Any variable changes have been validated to be backwards compatible

### CODE REVIEW

**Comment by JumaX** [comment]
@utkarsh5474 good catch.  Just updated to .7.

**Review by nsharma-git — APPROVED** [review]
LGTM
Out of curiosity, wondering if checks would be applicable in case of upgrade as well ?

**Review by nsharma-git — CHANGES_REQUESTED** [review]
Out of curiosity, wondering if checks would be applicable in case of upgrade as well ?

File: roles/variables/defaults/main.yml [hunk 1]
```diff
@@ -69,6 +69,9 @@ kerberos_kafka_broker_primary: "{{ (hostvars[ groups['kafka_broker'][0] | defaul
 
 open_file_limit: 500000
 
+### Variable to define minimum free disk space ratio for installation.
```
Comment (chromy96) [inline roles/variables/defaults/main.yml:72]: The documentation/comment and the variable name are clashing in my mind. The variable is a maximum used disk space, while the documentation mentions free space. I'd suggest naming it something like `max_used_disk_ratio` to make it easier to grasp what is it that is being defined

**Comment by sverrehu** [comment]
I don't understand the rationale behind this PR. The intro states _"validate that there is enough free disk space [...] to install the product."_ But the check is based on a ratio only, meaning that an attempt at installing on a 1 kb empty disk will pass this test and faile due to unsufficient disk space later.

File: roles/variables/defaults/main.yml (hunk 1 above)
Comment (JumaX) [reply roles/variables/defaults/main.yml:72]: I disagree.  To me, the way you describe it is more confusing. The variable is the ratio of free space that is required, which is what the documentation clearly states.

Comment (chromy96) [reply roles/variables/defaults/main.yml:72]: You're right, I have misread the code, thanks. 

I would still suggest a more descriptive name like `free_disk_space_ratio` or similar, as the name `disk_ratio` in the inventory tells very little what it is about (if overridden in the inventory, ofc)

Comment (JumaX) [reply roles/variables/defaults/main.yml:72]: That's fair, let me update that.

File: VARIABLES.md [hunk 2]
```diff
@@ -140,11 +140,11 @@ Default:  true
 
 ***
 
-### disk_ratio
+### free_disk_space
```
Comment (sverrehu) [inline VARIABLES.md:143]: Would `required_disk_space`, or perhaps even `required_disk_space_mb` make this variable more understandable without refering to the docs?

Comment (erikgb) [reply VARIABLES.md:143]: Agreed, at least a unit postfix should be included in the name.

**Review by nsharma-git — APPROVED** [review]

**Review by nsharma-git — APPROVED** [review]

**Pull request merged**